src/wifinode.cpp

```diff
@@ -52,3 +54,18 @@ void WifiNode::sendFragmentationNeeded(ipv4_headers_t* sourcePacket) {
   pbuf_free(pbuff);
   raw_remove(pcb);
 }
+
+void WifiNode::sendPacket(ipv4_headers_t* headers,
+                          void* payload,
+                          size_t length) {
+  struct raw_pcb* protocolControlBlock = raw_new(headers->protocol);
```

Comment:
What if the `headers->protocol` value is different than ICMP, TCP or UDP?

Reply:
It is a good question! I am thinking of a solution to this problem

Reply:
It's already checked:
 if (headers->protocol != IP_PROTO_ICMP && headers->protocol != IP_PROTO_UDP &&
      headers->protocol != IP_PROTO_TCP) {
    return;
  }